### PR TITLE
Scala3doc: Fix searchbar to match only by name

### PR DESCRIPTION
**Pull request by pikinier20**
It's fix of a bug reported in [#200](https://github.com/lampepfl/scala3doc/issues/200)

### CODE REVIEW

File: scala3doc/src/dotty/renderers/ScalaSearchbarDataInstaller.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.jetbrains.dokka.base.signatures.SignatureProvider
+import org.jetbrains.dokka._
+import org.jetbrains.dokka.model._
+import scala.collection.concurrent.TrieMap
+
+class ScalaSearchbarDataInstaller(val ctx: DokkaContext) extends SearchbarDataInstaller:
+
+  case class PageEntry(val name: String, val signature: String, val link: String, val pkg: String)
+
+  val pages = TrieMap[String, PageEntry]()
```
Comment (romanowski) [inline scala3doc/src/dotty/renderers/ScalaSearchbarDataInstaller.scala:24]: Maybe we can add a comment that this is such a strange piece of code (concurrent collection, mutable state etc.) due to dokka?

File: scala3doc/src/dotty/renderers/ScalaSearchbarDataInstaller.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  val signatureProvider = querySingle[DokkaBase, SignatureProvider](ctx, _.getSignatureProvider)
+
+  override def processPage(page: ContentPage, link: String) =
+    Option(page.getDocumentable) match {
+      case Some(member) => processMember(member, link)
+      case None => ()
```
Comment (romanowski) [inline scala3doc/src/dotty/renderers/ScalaSearchbarDataInstaller.scala:31]: We need to add support for titles from static documentation nodes

**Review by romanowski — CHANGES_REQUESTED** [review]

File: scala3doc/src/dotty/renderers/ScalaSearchbarDataInstaller.scala (hunk 1 above)
Comment (pikinier20) [reply scala3doc/src/dotty/renderers/ScalaSearchbarDataInstaller.scala:27]: added

File: scala3doc/src/dotty/renderers/ScalaHtmlRenderer.scala [hunk 3]
```diff
@@ -40,6 +41,19 @@ class SignatureRenderer(pageContext: ContentPage, sourceSetRestriciton: JSet[Dis
 
 class ScalaHtmlRenderer(ctx: DokkaContext) extends HtmlRenderer(ctx) {
 
+  // TODO #239
+  val hackScalaSearchbarDataInstaller: SearchbarDataInstaller = {
+    import java.lang.reflect.Field;
```
[the remark this answers is not in the document]
Comment (pikinier20) [reply scala3doc/src/dotty/renderers/ScalaHtmlRenderer.scala]: fixed

File: scala3doc/src/dotty/renderers/ScalaSearchbarDataInstaller.scala (hunk 2 above)
Comment (pikinier20) [reply scala3doc/src/dotty/renderers/ScalaSearchbarDataInstaller.scala]: added support

File: scala3doc/src/dotty/dokka/compat.scala [hunk 4]
```diff
@@ -57,3 +59,12 @@ extension [V] (map: JMap[SourceSetWrapper, V]):
 extension [V](jlist: JList[V]):
   def ++ (other: JList[V]): JList[V] =
     Stream.of(jlist, other).flatMap(_.stream).collect(Collectors.toList())
+
+object PluginUtils:
+    import scala.reflect.ClassTag
+    import scala.reflect._
+    def query[T <: DokkaPlugin, E](ctx: DokkaContext, queryFunction: (T) => ExtensionPoint[E])(implicit tag: ClassTag[T]): List[E] =
```
[the remark this answers is not in the document]
Comment (pikinier20) [reply scala3doc/src/dotty/dokka/compat.scala]: looks better

**Review by romanowski — APPROVED** [review]

**Comment by romanowski** [comment]
@smarter it seems that `dotty.tools.dotc.CompilationTests` is flaky on Windows (or jvm 8). Is that a known problem?

**Comment by smarter** [comment]
> @smarter it seems that dotty.tools.dotc.CompilationTests is flaky on Windows (or jvm 8). Is that a known problem?

Not that I know of, /cc @liufengyun

**Pull request merged**